tests/unit/conftest.py

```diff
@@ -56,24 +56,28 @@ def inner(reg: Optional[str] = None) -> Path:
 
 
 @pytest.fixture
-def cycling_mode(monkeypatch: pytest.MonkeyPatch):
-    """Set the Cylc cycling mode.
-
+def cycling_type(monkeypatch: pytest.MonkeyPatch):
+    """Initialize the cycling type according to cycling mode.
     Args:
         mode: The cycling mode.
-        time_zone: If using ISO8601/datetime cycling mode, you can specify a
+        time_zone: If using ISO8601/datetime cycling type, you can specify a
             custom time zone to use.
     """
-    def _cycling_mode(
-        mode: str = INTEGER_CYCLING_TYPE, time_zone: Optional[str] = None
-    ) -> None:
+    def _cycling_type(
+        mode: str = 'integer', time_zone: Optional[str] = None
+    ) -> str:
+        if mode == 'integer':
+            ctype = INTEGER_CYCLING_TYPE
+        else:
+            ctype = ISO8601_CYCLING_TYPE
```

Comment:
True, but I was  trying to maintain the distinction between the `cycling mode` user config item, and `cycling type`, which is internal.  And your parameterized tests mostly do this via a mocked config dict, so they should really specify cycling mode, not type ... and then this fixture sets the internal type according to the mode.  Finally, the `gregorian` cycling mode is the default, so I set those values to empty string rather than hard-wiring another magic value.

So how about we leave this PR as-is, and as a low priority follow-up we could refactor the tests to specify cycling type directly, but separately from the mocked config?